### PR TITLE
[CHK-449] Add scheduled task for payment instrument update

### PR DESCRIPTION
**Pull request by MarcoAGreco**
#### List of Changes
- Add scheduled task for payment instruments update
- Add `api-config` client generation

#### How Has This Been Tested?

 - Build
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by giovanniberti — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]

File: src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/PspFactory.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    @Autowired
+    private PspRepository pspRepository;
+
+    @AggregateFactory(Psp.class)
+    public Mono<Psp> newPsp(PspCode pspCode, PaymentInstrumentID paymentInstrumentID, PspStatus pspStatus,
+                            PspBusinessName pspBusinessName, PspBrokerName pspBrokerName,
+                            PspDescription pspDescription, PspPaymentInstrumentType pspPaymentInstrumentType) {
+
+        return pspRepository.findByPspCode(pspCode.value()).hasElements()
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/PspFactory.java:27]: What about:

```suggestion
        return pspRepository
            .findByPspCode(pspCode.value())
            .switchIfEmpty(Mono.error(pspAlreadyInUseException(pspCode)))
            .map(psp -> { ... })
```

File: src/main/java/it/pagopa/ecommerce/payment/instruments/domain/valueobjects/Psp.java [hunk 2]
```diff
@@ -13,13 +13,13 @@
 @Getter
 public class Psp implements Serializable {
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/payment/instruments/domain/valueobjects/Psp.java:14]: I was thinking about the naming of this class: what do you think about `PspSummary`? The thing that doesn't sit well with me it's the fact that we have both an aggregate and a value object with the same name, but maybe I'm just worrying too much over nothing

File: src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/Psp.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        this.pspStatus = pspStatus;
+        this.pspBusinessName = pspBusinessName;
+        this.pspBrokerName = pspBrokerName;
+        this.pspDescription = pspDescription;
+        this.pspLangagues = new ArrayList();
+        this.pspRanges = new ArrayList();
+        this.pspPaymentInstrumentType = pspPaymentInstrumentType;
+    }
+
+    public void enablePsp(PspStatus pspStatus) {
```
[the remark this answers is not in the document]
Comment (MarcoAGreco) [reply src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/Psp.java:56]: I've used same signature used for PaymentInstrument aggregate [here](https://github.com/pagopa/pagopa-ecommerce-payment-instruments-service/blob/44423ac8481007aa16fc9032a03edcc66987ee74/src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/PaymentInstrument.java#L40), maybe we should update both?

File: src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/PspFactory.java (hunk 1 above)
Comment (MarcoAGreco) [reply src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/PspFactory.java]: Legit, going to update with this snippet 👍

File: src/main/java/it/pagopa/ecommerce/payment/instruments/domain/valueobjects/Psp.java (hunk 2 above)
Comment (MarcoAGreco) [reply src/main/java/it/pagopa/ecommerce/payment/instruments/domain/valueobjects/Psp.java:14]: That's a good point, maybe we can remove the value object, it was defined before the domain model refactoring. Now we have separate collection for PaymentInstruments and PSPs. Should we do it in this PR or another one?

Comment (giovanniberti) [reply src/main/java/it/pagopa/ecommerce/payment/instruments/domain/valueobjects/Psp.java:14]: If it's not too messy I think it's worth removing the value object in this PR. Although the threshold of "messy" should be pretty tight: there's already a lot going on in this PR and I think it's better to keep this clean that doing too much at once

File: src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/Psp.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (giovanniberti) [reply src/main/java/it/pagopa/ecommerce/payment/instruments/domain/aggregates/Psp.java:56]: Yeah, I think as it is now it's a bit misleading/unclear

**Pull request merged**